cclib/parser/gaussianparser.py

```diff
@@ -213,6 +216,229 @@ def extract(self, inputfile, line):
                 "package_version"
             ] = f"{self.YEAR_SUFFIXES_TO_YEARS[year_suffix]}+{revision}"
             self.metadata["platform"] = platform
+            self.set_attribute('run_date', run_date)
+
+        # Extract number of processors, memory, keywords line, solvation,
+        # empirical dispersion and type of system (ground or transition state)
+        # ******************************************
+        # %mem=24GB
+        # %nprocshared=12
+        # Will use up to   12 processors via shared memory.
+        # %chk=Int23.chk
+        # ----------------------------------------------------------------------
+        # # uM062X/Def2TZVP guess=read geom=check opt=(calcfc,ts,noeigen,maxstep
+        # =5) freq=noraman
+        # ----------------------------------------------------------------------
+        elif '%mem' in line:
+            mem = line.strip().split('=')[-1]
+            self.set_attribute('memory', mem)
+
+        elif '%nprocs' in line:
+            nprocs = int(line.strip().split('=')[-1])
+            self.set_attribute('processors', nprocs)
+
+        elif '#' in line and not hasattr(self, 'keywords_line'):
+            keywords_line = ''
+            while '----------' not in line:
+                keywords_line += line.rstrip("\n")[1:]
+                line = next(inputfile)
+                self.set_attribute('keywords_line', keywords_line[2:])
+            qm_solv,qm_disp = 'gas_phase','none'
+            calc_type = 'ground_state'
+            calcfc_found, ts_found = False, False
+            for keyword in keywords_line.split():
+                if keyword.lower().find('opt') > -1:
+                    if keyword.lower().find('calcfc') > -1:
+                        calcfc_found = True
+                    if keyword.lower().find('ts') > -1:
+                        ts_found = True
+                elif keyword.lower().startswith('scrf'):
+                    qm_solv = keyword
+                elif keyword.lower().startswith('emp'):
+                    qm_disp = keyword
+            if calcfc_found and ts_found:
+                calc_type = 'transition_state'
+
+            self.set_attribute('solvation', qm_solv)
+            self.set_attribute('dispersion', qm_disp)
+            self.set_attribute('calc_type', calc_type)
```

Comment:
Good point - it took me some time to figure this out. I ended up using this format since gen and genecp are important keywords in AQME. In Gaussian, the gen keyword cannot be tracked in the standard outputs (or I couldn't figure out how to do it).

`qm_solv` was removed
`qm_disp` was adapted to self.metadata["dispersion model"] (based on a previous comment)
`calc_type` was adapted to self.metadata["calc_type"] and removed from data.py in commit [a5772d6](https://github.com/cclib/cclib/pull/1269/commits/a5772d627a470ce49290ca82d59b7cb0f8745ee7)